tests/backup/backup_helper.go

```diff
@@ -1426,3 +1430,85 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
 	_, err := task.DoRetryWithTimeout(backupDeletionSuccessCheck, backupDeleteTimeout, backupDeleteRetryTime)
 	return err
 }
+
+// getStorkImageVersion returns current stork image version.
+func getStorkImageVersion() (string, error) {
+	storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+	if err != nil {
+		return "", err
+	}
+	storkImage := storkDeployment.Spec.Template.Spec.Containers[0].Image
+	storkImageVersion := strings.Split(storkImage, ":")[len(strings.Split(storkImage, ":"))-1]
+	return storkImageVersion, nil
+}
+
+// upgradeStorkVersion upgrades the stork to the provided version.
+func upgradeStorkVersion(storkImageToUpgrade string) error {
+	currentStorkImageStr, err := getStorkImageVersion()
+	if err != nil {
+		return err
+	}
+	currentStorkVersion, err := version.NewSemver(currentStorkImageStr)
+	if err != nil {
+		return err
+	}
+
+	storkImageVersionToUpgradeStr := strings.Split(storkImageToUpgrade, ":")[len(strings.Split(storkImageToUpgrade, ":"))-1]
+	storkImageVersionToUpgrade, err := version.NewSemver(storkImageVersionToUpgradeStr)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Current stork version : %s", currentStorkVersion)
+	log.Infof("Upgrading stork version to : %s", storkImageVersionToUpgrade)
+
+	if currentStorkVersion.GreaterThanOrEqual(storkImageVersionToUpgrade) {
+		log.Errorf("Cannot upgrade stork version from %s to %s as the current version is higher than the provided version", currentStorkVersion, storkImageVersionToUpgrade)
+		return nil
+	}
+
+	isOpBased, err := Inst().V.IsOperatorBasedInstall()
+	if err != nil {
+		return err
+	}
+	if isOpBased {
+		storageSpec, err := Inst().V.GetDriver()
+		if err != nil {
+			return err
+		}
+		storageSpec.Spec.Stork.Image = storkImageToUpgrade
+		_, err = operator.Instance().UpdateStorageCluster(storageSpec)
+		if err != nil {
+			return err
+		}
+	} else {
+		storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+		if err != nil {
+			return err
+		}
+
+		storkDeployment.Spec.Template.Spec.Containers[0].Image = storkImageToUpgrade
+		_, err = apps.Instance().UpdateDeployment(storkDeployment)
+		if err != nil {
+			return err
+		}
+	}
+
+	// validate stork pods after upgrade
```

Comment:
Add a check where we are ensuring all the stork pods are up and running

Reply:
ValidateDepoylment in below step verifies it.